docs/DirectoryManager.md

```diff
@@ -0,0 +1,127 @@
+
+Most VBA programs seem to not use `CurDir()` directly. They either do one of the following things -
+
+**Get absolute path using `ActiveWorkbook.Path`:**
+
+`FileName = ActiveWorkbook.Path & "\myfile.xlsx`
+
+> `ActiveWorkbook.path` is the parent directory of the current active workbook
+
+**Or set the current working directory to `ActiveWorkbook.Path`:**
+
+`ChDir (ActiveWorkbook.path)`
+
+> Note: This usage indicates that developers are not using the functionality of `CurDir()` directly. And t
+> hey are not using the persistence ability of the `CurDir()` API.
+
+> `CurDir("D")` returns the current working directory on `D Drive`. The current working drive itself can be changed by `ChDrive("D")`.
+
+### File Path
+
+File paths on windows start with a drive letter followed by a colon. The file separator is a backward slash. File and folde names are **case sensitive**.
+
+`C:\Users\Satvik\Desktop\Google`
+
+File paths on macOS doesn't have a drive letter. The file separator is a forward slash. File and folde names are **case insensitive**.
+
+`/Users/Satvik/Desktop/Google`
+
+VBA also support legacy Macintosh file paths. Apple has deprecated this type of file paths.
+
+`Macintosh HD:Users:Satvik:Desktop:Google`
+
+> For the most part, the main differences between mac and windows file paths are the drive letter and the file separators.
+
+### `ChDir(path)`
+
+The api is used to change the current working directory. The path could either be absolute path or relative path to current working directory.
+
+```
+ChDir ("C:\Users\Satvik\Desktop\Google") ' Absolute Path
+ChDir ("..\Documents") ' Relative Path
+```
+
+> ChDir() doesn't seem to work for relative paths in macOS
+
+## DirectoryManager Implementation in Apps Script
+
+Therefore a lot of thought has gone into the design of this module. The design aims to provide maximum support for Windows VBA API usage. Some legacy and deprecated API and features are not being implemented.
+
+### `CurDir()`
+
+Official Documentation - https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/curdir-function
+
+**VBA API Usage:**
+
+`CurDir [ (drive) ]`
+
+Design:
+
+1. `CurDir` is set to `ActiveWorkbook.path` in the beginning of each macro call. This behaviour differs from Windows but this has been decided as this is the most expected usage. 
+1. Changes to CurDir does not persist across calls
+1. If a Drive is passed to `CurDir` then, if the drive is same as current working directory, then just return it. If the drive passed is different, then just return root directory. For example `CurDir("E")`, return `E:\`
```

Comment:
At what point will you resolve "E" using file mapper?

Reply:
Fixed, I added the point